daemon/src/projection.rs

```diff
@@ -208,18 +220,76 @@ pub struct Cfd {
 /// - It serves as an aggregate that is hydrated from events.
 ///
 /// This dual-role motivates the existence of this struct.
-#[derive(Default, Clone, Debug)]
+#[derive(Clone, Debug)]
 struct Aggregated {
     /// If this is present, we have an active DLC.
     latest_dlc: Option<Dlc>,
     /// If this is present, it should have been published.
     collab_settlement_tx: Option<(Transaction, Script)>,
     /// If this is present, it should have been published.
     cet: Option<Transaction>,
-    closing_price: Option<Price>,
+
+    /// If this is present the cet has not been published
+    timelocked_cet: Option<Transaction>,
 
     commit_published: bool,
     refund_published: bool,
+
+    role: Role,
+}
+
+impl Aggregated {
+    fn new(role: Role) -> Self {
+        Self {
+            latest_dlc: None,
+            collab_settlement_tx: None,
+            cet: None,
+            timelocked_cet: None,
+            commit_published: false,
+            refund_published: false,
+            role,
+        }
+    }
+
+    fn payout(self) -> Option<Amount> {
+        if let Some((tx, script)) = self.collab_settlement_tx {
+            return Some(extract_payout_amount(tx, script));
+        }
+
+        match (self.cet, self.timelocked_cet) {
+            (Some(tx), _) | (_, Some(tx)) => {
+                let dlc = self
+                    .latest_dlc
+                    .as_ref()
+                    .expect("dlc to be present when we have a cet");
```

Comment:
This can use a `?` to bail if the DLC is not set.

Reply:
Only if I change the return type to be a `Result`. I think it is fair to assume that the `Dlc` is always set here.